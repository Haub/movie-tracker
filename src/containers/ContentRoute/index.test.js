import React from "react";
import { MemoryRouter } from "react-router";
import { shallow, mount, render } from "enzyme";

import { ContentRoute } from ".";

describe('ContentRoute container', () => {
  let wrapper;
  let displayHanksMovies;
  let movies;
  let favorites;

  beforeEach(() => {
    displayHanksMovies = jest.fn();
  })

  it('should match snapshot with the / path', () =>{
    wrapper = shallow(
      <ContentRoute 
        displayHanksMovies={displayHanksMovies} 
        movies={[]}
        favorites={[]}
      />);

    render(
      <MemoryRouter initialEntries={['/']}>
        <ContentRoute
          displayHanksMovies={displayHanksMovies}
          movies={[]}
          favorites={[]}
        />
      </MemoryRouter>
    ); 
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with the /favorites path', () =>{
    wrapper = shallow(
      <ContentRoute 
        displayHanksMovies={displayHanksMovies} 
        movies={[]}
        favorites={[]}
      />);

    render(
      <MemoryRouter initialEntries={['/favorites']}>
        <ContentRoute
          displayHanksMovies={displayHanksMovies}
          movies={[]}
          favorites={[]}
        />
      </MemoryRouter>
    ); 
    expect(wrapper).toMatchSnapshot();
  });

  it('should call displayHanksMovies on componentDidMount', () =>{
    let mockHanksMovies = {cast:[]}
    let mockFn = jest.fn().mockImplementation(() => (mockHanksMovies))
    window.fetch = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        json: () => Promise.resolve(mockHanksMovies)
      });
    });
    wrapper = shallow(
      <ContentRoute 
        displayHanksMovies={mockFn} 
        movies={[]}
        favorites={[]}
      />);
    render(
      <MemoryRouter initialEntries={['/']}>
        <ContentRoute
          displayHanksMovies={mockFn}
          movies={[]}
          favorites={[]}
        />
      </MemoryRouter>
    ); 
    wrapper.instance().componentDidMount();
    expect(mockFn).toHaveBeenCalled();
  });
})