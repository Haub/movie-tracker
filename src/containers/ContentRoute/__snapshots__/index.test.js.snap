// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentRoute container should match snapshot with the / path 1`] = `
<Switch>
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    path="/"
    render={[Function]}
  />
</Switch>
`;

exports[`ContentRoute container should match snapshot with the /favorites path 1`] = `
<Switch>
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    path="/"
    render={[Function]}
  />
</Switch>
`;
